components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -1962,4 +1968,70 @@ public static boolean isSCIM2UserMaxItemsPerPageEnabled() {
         }
         return Boolean.parseBoolean(scim2UserMaxItemsPerPageEnabledProperty);
     }
+
+    /**
+     * Validates the signature of the given data for the specified tenant domain.
+     *
+     * @param data         The data to be verified.
+     * @param signature    The signature to be verified.
+     * @param tenantDomain The tenant domain to which the data belongs.
+     * @return true if the signature is valid, false otherwise.
+     * @throws IdentityKeyStoreResolverException If an error occurs during the signature validation process.
+     */
+    public static boolean validateSignature(String data, byte[] signature, String tenantDomain)
+            throws IdentityKeyStoreResolverException {
+
+        try {
+            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
```

Comment:
We can take this out of the try block

Reply:
Fixed by - https://github.com/wso2/carbon-identity-framework/pull/6176/commits/43523e575a591197e6f3b04210f9be0a83ee4fe0